trimChamberV3.py

```diff
@@ -57,29 +57,64 @@
         exit(os.EX_USAGE)
         pass
 
-    if options.calFileARM is None:
-        print("You must provide the calibration for the CFG_THR_ARM_DAC register")
-        print("Please relaunch with the --calFileARM argument")
-        exit(os.EX_USAGE)
-        pass
-
     # Get the calibration for the CFG_CAL_DAC register
     tuple_calInfo = parseCalFile(options.calFileCAL)
     calDac2Q_Slope = tuple_calInfo[0]
     calDac2Q_Intercept = tuple_calInfo[1]
 
-    # Get the calibration for the CFG_THR_ARM_DAC register
-    tuple_calInfo = parseCalFile(options.calFileARM)
-    thrArmDac2Q_Slope = tuple_calInfo[0]
-    thrArmDac2Q_Intercept = tuple_calInfo[1]
+    dict_func_scurveMeanVsThrDac = {}
 
-    chMin = options.chMin
-    chMax = options.chMax + 1
+    import ROOT as r
 
     # Make the ohKey
     ohKey = (options.shelf,options.slot,options.gtx)
 
-    if options.dirPath == None: 
+    # Get the calibration for the CFG_THR_ARM_DAC register    
+    if options.calFileARM is None:
+        envCheck('DATA_PATH')
+        dataPath = os.getenv('DATA_PATH')
+        calFileBase = "{0}/{1}/calFile_CFG_THR_ARM_DAC_{1}".format(dataPath,chamber_config[ohKey])
+        if os.path.isfile(calFileBase + ".root"):
+            TFile_calInfo = r.TFile(calFileBase + ".root")
+            for vfat in range(0,24):
+                for key in TFile_calInfo.GetDirectory("VFAT{0}".format(vfat)).GetListOfKeys():
+                    if "func_ScurveMean_vs_CFG_THR_ARM_DAC_" in key.GetName() and "_vfatN{0}_".format(vfat) in key.GetName():
+                        dict_func_scurveMeanVsThrDac[vfat] = TFile_calInfo.GetDirectory("VFAT{0}".format(vfat)).Get(key.GetName())
+                        break
+        elif os.path.isfile(calFileBase + ".txt"):            
+            tuple_calInfo = parseArmDacCalFile(calFileBase + ".txt")
+            for vfat in range(0,24):
+                dict_func_scurveMeanVsThrDac[vfat] = r.TF1("func_scurveMeanVsThrDac_vfat%d"%(vfat),"[0]*x^4+[1]*x^3+[2]*x^2+[3]*x+[4]")
+                dict_func_scurveMeanVsThrDac[vfat].SetParameter(0,tuple_calInfo[0][vfat])
+                dict_func_scurveMeanVsThrDac[vfat].SetParameter(1,tuple_calInfo[1][vfat])
+                dict_func_scurveMeanVsThrDac[vfat].SetParameter(2,tuple_calInfo[2][vfat])
+                dict_func_scurveMeanVsThrDac[vfat].SetParameter(3,tuple_calInfo[3][vfat])
+                dict_func_scurveMeanVsThrDac[vfat].SetParameter(4,tuple_calInfo[4][vfat])
```

Comment:
Lines 87 to 92 can be simplified heavily using a `lambda`:

```python
dac2Charge[vfat] = lambda dac, vfat: sum([ myCalInfo[x][vfat]*pow(dac,4-x) for x in range(0,5) ])
```

Reply:
The `lambda` could be stored in the dict as well, e.g.:

 ```python
dict_func_scurveMeanVsThrDac[vfat] = lambda dac, vfat: sum([ myCalInfo[x][vfat]*pow(dac,4-x) for x in range(0,5) ])
charge = dict_func_scurveMeanVsThrDac[vfat](dacVal,vfat)
```

Reply:
But looking at the code below it would be too hard to toggle between a `lambda` and a `TF1`.  So just leave it as is.